accord-core/src/main/java/accord/utils/RelationMultiMap.java

```diff
@@ -221,7 +221,7 @@ public T build()
             {
                 sortedKeyIndexes = new int[keyCount];
                 sortedKeys = Arrays.copyOf(keys, keyCount);
-                Arrays.sort(sortedKeys);
+                Arrays.sort(sortedKeys, adapter.keyComparator());
```

Comment:
I feel that there are more cases, these were just the cases needed to make the test stable...  I tried a patch to make this class typed to make sure we are the proper `Comparable` but that patch got very large, so punted for this work around... I do feel that range should be `Comparable<Range>` instead and just offer a `compareToKey`